src/sentry/monitors/tasks.py

```diff
@@ -303,11 +304,13 @@ def mark_environment_missing(monitor_environment_id: int, ts: datetime):
     # Specifically important for interval where it's a function of some
     # starting time.
     #
-    # Trim tzinfo. This is always UTC. ts does not have tzinfo and
-    # get_prev_schedule will fail if the start and reference time have
-    # different tzinfo.
-    start_ts = expected_time.replace(tzinfo=None)
-    most_recent_expected_ts = get_prev_schedule(start_ts, ts, monitor.schedule)
+    # When computing our timestamps MUST be in the correct timezone of the
+    # monitor to compute the previous schedule
+    most_recent_expected_ts = get_prev_schedule(
+        expected_time.astimezone(monitor.timezone),
+        ts.astimezone(monitor.timezone),
+        monitor.schedule,
+    )
 
     mark_failed(checkin, ts=most_recent_expected_ts)
```

Comment:
just to confirm, this gets stored in `UTC` even when we pass it with the timezone attached, right?

Reply:
Yeah Postgres always stores things as UTC